pytest_plugins/issue_handlers.py

```diff
@@ -0,0 +1,272 @@
+    )
+
+
+@pytest.hookimpl(trylast=True)
+def pytest_collection_modifyitems(session, items, config):
+    """Generate the issue collection (using bz cache via bugzilla issue handler util"""
+    # This collection includes pre-processed `is_open` status for each issue
+    # generate_issue_collection will save a file, set by --bz-cache value, on each run.
+    pytest.issue_data = generate_issue_collection(items, config)
+
+    # No item deselect, just adding skipif markers
+    for item in items:
+        skip_if_open = item.get_closest_marker('skip_if_open')
+        if skip_if_open:
+            # marker must have `BZ:123456` as argument.
+            issue = skip_if_open.kwargs.get('reason') or skip_if_open.args[0]
+            item.add_marker(pytest.mark.skipif(is_open(issue), reason=issue))
+
+
+IS_OPEN = re.compile(
+    # To match `if is_open('BZ:123456'):`
+    r"\s*if\sis_open\(\S(?P<src>\D{2})\s*:\s*(?P<num>\d*)\S\)\d*"
+)
+
+NOT_IS_OPEN = re.compile(
+    # To match `if not is_open('BZ:123456'):`
+    r"\s*if\snot\sis_open\(\S(?P<src>\D{2})\s*:\s*(?P<num>\d*)\S\)\d*"
+)
+
+COMPONENT = re.compile(
+    # To match :CaseComponent: FooBar
+    r"\s*:CaseComponent:\s*(?P<component>\S*)",
+    re.IGNORECASE,
+)
+
+IMPORTANCE = re.compile(
+    # To match :CaseImportance: Critical
+    r"\s*:CaseImportance:\s*(?P<importance>\S*)",
+    re.IGNORECASE,
+)
+
+BZ = re.compile(
+    # To match :BZ: 123456, 456789
+    r"\s*:BZ:\s*(?P<bz>.*\S*)",
+    re.IGNORECASE,
+)
+
+
+def generate_issue_collection(items, config):  # pragma: no cover
+    """Generates a dictionary with the usage of Issue blockers
+
+    For use in pytest_collection_modifyitems hook
+
+    Arguments:
+        items {list} - List of pytest test case objects.
+        config {dict} - Pytest config object.
+
+    Returns:
+        [List of dicts] - Dicts indexed by "<handler>:<issue>"
+
+        Example of return data::
+
+            {
+                "XX:1625783" {
+                    "data": {
+                        # data taken from REST api,
+                        "status": ...,
+                        "resolution": ...,
+                        ...
+                        # Calculated data
+                        "is_open": bool,
+                        "is_deselected": bool,
+                        "clones": [list],
+                        "dupe_data": {dict}
+                    },
+                    "used_in" [
+                        {
+                            "filepath": "tests/foreman/ui/test_sync.py",
+                            "lineno": 124,
+                            "testcase": "test_positive_sync_custom_ostree_repo",
+                            "component": "Repositories",
+                            "usage": "skip_if_open"
+                        },
+                        ...
+                    ]
+                },
+                ...
+            }
+    """
+    settings.configure()
```

Comment:
Can't assume any other plugins are going to load the settings, otherwise I'm setting this up for failure.

Reply:
This is more a failure of the overall design of the settings instance as it exists right now. No module should have to trigger configure, because no module can be certain it hasn't already been done - and no module should have to check whether the configuration file has been read yet.

Reply:
@mshriver Agree on the failure of design, but it is how it is :)

Is DynaConf going to help here ?

Reply:
Note I did update the use to include the check first.

Dynaconf will indeed improve this, as modules reading settings will no longer have to worry about whether or not the config has been loaded yet.